### ISSUE TITLE
Can't compile

### ISSUE THREAD

**Issue by Csokis** [flightlessmango/MangoHud#759]
I can't compile on Arch Linux (5.17.6-arch1-1).

> ╰─$ ./build.sh build                                                                            1 ↵
Command: build
ninja: Entering directory `build/meson64'
[2/5] Compiling C++ object subprojects/spdlog-1.8.5/src/libspdlog.a.p/fmt.cpp.o
FAILED: subprojects/spdlog-1.8.5/src/libspdlog.a.p/fmt.cpp.o
ccache c++ -Isubprojects/spdlog-1.8.5/src/libspdlog.a.p -Isubprojects/spdlog-1.8.5/src -I../../subprojects/spdlog-1.8.5/src -I../../subprojects/spdlog-1.8.5/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Werror -std=c++14 -O3 -fPIC -pthread -DSPDLOG_COMPILED_LIB -MD -MQ subprojects/spdlog-1.8.5/src/libspdlog.a.p/fmt.cpp.o -MF subprojects/spdlog-1.8.5/src/libspdlog.a.p/fmt.cpp.o.d -o subprojects/spdlog-1.8.5/src/libspdlog.a.p/fmt.cpp.o -c ../../subprojects/spdlog-1.8.5/src/fmt.cpp
In file included from ../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format-inl.h:28,
                 from ../../subprojects/spdlog-1.8.5/src/fmt.cpp:10:
In member function ‘constexpr void fmt::v7::detail::fill_t<Char>::operator=(fmt::v7::basic_string_view<Char>) [with Char = char]’,
    inlined from ‘constexpr void fmt::v7::detail::specs_setter<Char>::on_fill(fmt::v7::basic_string_view<Char>) [with Char = char]’ at ../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:2445:17,
    inlined from ‘constexpr const Char* fmt::v7::detail::parse_align(const Char*, const Char*, Handler&&) [with Char = char; Handler = specs_checker<specs_handler<fmt::v7::basic_format_parse_context<char>, fmt::v7::basic_format_context<std::back_insert_iterator<buffer<char> >, char> > >&]’ at ../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:2823:24,
    inlined from ‘constexpr const Char* fmt::v7::detail::parse_format_specs(const Char*, const Char*, SpecHandler&&) [with Char = char; SpecHandler = specs_checker<specs_handler<fmt::v7::basic_format_parse_context<char>, fmt::v7::basic_format_context<std::back_insert_iterator<buffer<char> >, char> > >&]’ at ../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:2883:22:
../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:1158:48: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1158 |     for (size_t i = 0; i < size; ++i) data_[i] = s[i];
      |                                       ~~~~~~~~~^~~~
../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h: In function ‘constexpr const Char* fmt::v7::detail::parse_format_specs(const Char*, const Char*, SpecHandler&&) [with Char = char; SpecHandler = specs_checker<specs_handler<fmt::v7::basic_format_parse_context<char>, fmt::v7::basic_format_context<std::back_insert_iterator<buffer<char> >, char> > >&]’:
../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:1148:8: note: at offset 4 into destination object ‘fmt::v7::detail::fill_t<char>::data_’ of size 4
 1148 |   Char data_[max_size] = {Char(' '), Char(0), Char(0), Char(0)};
      |        ^~~~~
In member function ‘constexpr void fmt::v7::detail::fill_t<Char>::operator=(fmt::v7::basic_string_view<Char>) [with Char = char]’,
    inlined from ‘constexpr void fmt::v7::detail::specs_setter<Char>::on_fill(fmt::v7::basic_string_view<Char>) [with Char = char]’ at ../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:2445:17,
    inlined from ‘constexpr const Char* fmt::v7::detail::parse_align(const Char*, const Char*, Handler&&) [with Char = char; Handler = specs_checker<specs_handler<fmt::v7::basic_format_parse_context<char>, fmt::v7::basic_format_context<buffer_appender<char>, char> > >&]’ at ../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:2823:24,
    inlined from ‘constexpr const Char* fmt::v7::detail::parse_format_specs(const Char*, const Char*, SpecHandler&&) [with Char = char; SpecHandler = specs_checker<specs_handler<fmt::v7::basic_format_parse_context<char>, fmt::v7::basic_format_context<buffer_appender<char>, char> > >&]’ at ../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:2883:22:
../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:1158:48: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1158 |     for (size_t i = 0; i < size; ++i) data_[i] = s[i];
      |                                       ~~~~~~~~~^~~~
../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h: In function ‘constexpr const Char* fmt::v7::detail::parse_format_specs(const Char*, const Char*, SpecHandler&&) [with Char = char; SpecHandler = specs_checker<specs_handler<fmt::v7::basic_format_parse_context<char>, fmt::v7::basic_format_context<buffer_appender<char>, char> > >&]’:
../../subprojects/spdlog-1.8.5/include/spdlog/fmt/bundled/format.h:1148:8: note: at offset 4 into destination object ‘fmt::v7::detail::fill_t<char>::data_’ of size 4
 1148 |   Char data_[max_size] = {Char(' '), Char(0), Char(0), Char(0)};
      |        ^~~~~
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.

**Comment by jackun** [comment]
Try again

**Comment by Csokis** [comment]
Unfortunately the same error! :sweat:

**Comment by flightlessmango** [comment]
Have you tried running a `./build.sh clean` before?

**Comment by Csokis** [comment]
No. :roll_eyes: Now it is. :blush: It works! :sunglasses: Thank You! :thumbsup:
